presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java

```diff
@@ -377,8 +377,8 @@ public void testOutputStats()
         assertEquals(statistics.getAnalysisTime().get().toMillis(), queryStats.getAnalysisTime().toMillis());
         assertEquals(statistics.getExecutionTime().get().toMillis(), queryStats.getExecutionTime().toMillis());
         assertEquals(statistics.getPeakUserMemoryBytes(), queryStats.getPeakUserMemoryReservation().toBytes());
-        assertEquals(statistics.getPeakTotalNonRevocableMemoryBytes(), queryStats.getPeakTaskUserMemory().toBytes());
-        assertEquals(statistics.getPeakTaskTotalMemory(), queryStats.getPeakTaskUserMemory().toBytes());
+        assertEquals(statistics.getPeakTotalNonRevocableMemoryBytes(), queryStats.getPeakTotalMemoryReservation().toBytes());
```

Comment:
see: `io/prestosql/execution/QueryStateMachine.java:120`
```
    // peak of the user + system + revocable memory reservation
```
so `getPeakTotalNonRevocableMemoryBytes()` != `getPeakTotalMemoryReservation`

We don't track `PeakTotalNonRevocableMemoryBytes` in `QueryStateMachine`

Reply:
I see. But we seem to set `getPeakTotalNonRevocableMemoryBytes` from `getPeakTotalMemoryReservation`. Is it contradicting with the comment in `QueryStateMachine`?

https://github.com/prestosql/presto/blob/37c74ad15f5f0c9c770c2aa491e49ffe5146d4b1/presto-main/src/main/java/io/prestosql/event/QueryMonitor.java#L243

Reply:
This is a bug. We should either:
1. have a dedicated counter for non-revocable reservations
2. Change field name in `QueryStatistics`

I think we should do 1) for backward compatibility reason.

Reply:
For the bug fix, it's worth creating a separate PR. (People tend to pay less attention to "Fix flaky ..." PRs.)

Reply:
Okay, I'll work on that.

Reply:
I would disable this particular check and add a `TODO` (possibly pointing to Github issue).

Reply:
Okay, I filed an issue https://github.com/prestosql/presto/issues/4060